### ISSUE TITLE
`FormCollection` model binding ignores the current prefix

### ISSUE THREAD

**Issue by dougbu** [aspnet/Mvc#2331]
for example, create an action with a parameter such as `[FromForm(Name = "prefix")] FormCollection collection`.  the parameter binds to a collection containing the entire form content, not just the `prefix` value (if any).  this raises questions such as "why is this supported?" and "why is the user's intention ignored?"

similarly, create an action with a parameter of the following type:

``` c#
public class WrapperModel
{
  ...
  public FormCollection TheRest { get; set; }
}
```

the `WrapperModel.TheRest` property always references a collection containing the entire form content after binding.  the collection does _not_ exclude other `WrapperModel` property values.

suggest binding a `FormCollection` should either filter the collection base on an explicitly-provided (e.g. with attributes or containing model classes) prefix or not bind at all if an explicitly-provided prefix exists.

TL;DR
while ignoring the prefix could plausibly lead to scenarios where the model binding system does not correctly fall back to the empty prefix, in practise this doesn't appear to be an issue.  a top-level `FormCollection` parameter consistently binds using the initial (parameter name or specified) prefix.  binding a `FormCollection` property isn't attempted using the initial prefix unless a value provider contains a matching prefix for that property or a sibling; accidental matches are highly unlikely.

**Comment by rynowak** [comment]
This would be a breaking change to do now and there have been no requests for something like this.

Suggest closing

/cc @Eilon
